wcfsetup/install/files/lib/system/service/worker/Util.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace wcf\system\service\worker;
+
+use Jose\Component\Core\JWK;
+use ParagonIE\ConstantTime\Base64UrlSafe;
+use ParagonIE\ConstantTime\Binary;
+
+/**
+ * @author      Olaf Braun
+ * @copyright   2001-2024 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class Util
+{
+    /**
+     * Unserialize the public key into its x and y components.
+     *
+     * @param string $data
+     * @return array{x: string, y: string}
+     */
+    public static function unserializePublicKey(string $data): array
+    {
+        $data = \bin2hex($data);
+        if (Binary::safeSubstr($data, 0, 2) !== '04') {
+            throw new \InvalidArgumentException('Invalid public key format');
+        }
+        $data = Binary::safeSubstr($data, 2);
+        $dataLength = Binary::safeStrlen($data);
+
+        return [
+            'x' => \hex2bin(Binary::safeSubstr($data, 0, $dataLength / 2)),
+            'y' => \hex2bin(Binary::safeSubstr($data, $dataLength / 2)),
+        ];
+    }
+
+    /**
+     * Serialize the public key.
+     *
+     * @param string $x
+     * @param string $y
+     * @return string
+     */
+    public static function serializePublicKey(string $x, string $y): string
+    {
+        $hexString = '04';
+        $hexString .= \str_pad(\bin2hex(Base64UrlSafe::decode($x, true)), 64, '0', STR_PAD_LEFT);
+        return $hexString . \str_pad(\bin2hex(Base64UrlSafe::decode($y, true)), 64, '0', STR_PAD_LEFT);
```

Comment:
Likewise this probably should return the binary representation, as `ServiceWorkerHandler` decodes the hexadecimal representation to binary anyway. Cryptography is much simpler when dealing with the binary representation.